include/lsst/sphgeom/Pixelization.h

```diff
@@ -23,6 +23,11 @@
 #ifndef LSST_SPHGEOM_PIXELIZATION_H_
 #define LSST_SPHGEOM_PIXELIZATION_H_
 
+// PyInt_GetMax is not in the python 3 C API, so we define it for python >=3
+#if PY_MAJOR_VERSION >= 3
+#define PyInt_GetMax(x) 2<<31
+#endif
```

Comment:
First of all, I think this should be addressed in python/lsst/sphgeom/pixelizations.i - adding Python C-API stand-ins to our C++ library headers feels like pollution to me.

Secondly: I would recommend trying to replace [this line](https://github.com/lsst/sphgeom/blob/tickets/DM-7311/python/lsst/sphgeom/pixelizations.i#L44) with:

    if (p <= static_cast<unsigned long long>(std::numeric_limits<long>::max())) {

which should be equivalent (based on my reading of the docs for `PyInt_GetMax`).